### PR TITLE
Make Attackby (etc) parameters use event args.

### LINKED ISSUE #74: Make Attackby (etc) parameters use event args.
@clusterfack 

Right now they're just direct arguments like in BYOND, but this means adding new parameters is a breaking change and causes refactors. Would be nice if they were event args.

### PR DESCRIPTION
**Pull request by PrPleGoo**
AttackBy with an EventArg object for a parameter
AttackHand with an EventArg object for a parameter
AttackByEventArgs inherits from EventArgs again
RangedAttackBy with an EventArg object for a parameter
AfterAttack with an EventArg object for a parameter
UseEntity with an EventArg object for a parameter
Activate with an EventArg object for a parameter
Fixes #74

### CODE REVIEW

**Comment by PrPleGoo** [comment]
It also includes some inconsistent naming of the involved interfaces, their methods and their parameters.

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [submodule "RobustToolbox"]
 	path = RobustToolbox
-	url = https://github.com/space-wizards/RobustToolbox.git
+	url = https://github.com/PrPleGoo/RobustToolbox.git
```
Comment (its-sprite) [inline .gitmodules]: bruh sound effect 2

Comment (PrPleGoo) [reply .gitmodules]: Thanks, Fixed

**Comment by clusterfack** [comment]
Quality Work

**Pull request merged**